pgmpy/base/base.py

```diff
@@ -0,0 +1,248 @@
+import inspect
+from abc import ABC, abstractmethod
+from typing import Hashable, Iterable, Optional
+
+import networkx as nx
+
+from pgmpy.base._mixin_roles import _GraphRolesMixin
+
+
+class _CoreGraphABC(ABC):
```

Comment:
Wouldn't suggest using `ABC`. It is too strict and creates dependency between the ABC class and the class inheriting it.